manager/apis/app/v1alpha1/m4dapplication_types.go

```diff
@@ -8,7 +8,57 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// DataContext indicates data set chosen by the Data Scientist to be used by his application,
+// RecurrenceType defines how often the data loads should be repeated
+// Relevant to implicit and explicit copies, streamed read flows
+type RecurrenceType string
+
+// RecurrenceType valid values
+const (
+	None    RecurrenceType = "none" // load once, and no repeat
+	Daily   RecurrenceType = "daily"
+	Weekly  RecurrenceType = "weekly"
+	Monthly RecurrenceType = "monthly"
+	Stream  RecurrenceType = "stream"
+)
+
+// DataResidency indicates the geography in which the data was created
+// TODO: Values should come from a taxonomy
+type DataResidency string
+
+// DataSensitivity indicates the level of sensitivity of the data in the data set
+// TODO: Values should be taken from a taxonomy
+type DataSensitivity string
+
+// ExternalDataContext provides the information for reading or writing data that is not cataloged in the data catalog.
+// In this case the metadata has to be declared, rather than obtained from the data catalog.
+type ExternalDataContext struct {
+	// External data source not cataloged in the data catalog
+	// +required
+	ExternalStore DataStore `json:"externalStore"`
+
+	// Residency indicates the geographical origin of the data, especially important if there is an personal or other sensitive data.
+	// Assumes all data in the data set is from the same geography.
+	// TODO: In the future perhaps we need to handle situations where there are multiple residencies
+	// +required
+	Residency DataResidency `json:"residency"`
+
+	// Sensitivity indicates if there is any sensitive data in the data set.
+	// Since there may be different types of data with different levels of sensitivity,
+	// this is a list.  (ex: confidential, personal, special personal information)
+	// No values means there is no sensitive data.
+	// +optional
+	Sensitivity []DataSensitivity `json:"sensitivity,omitempty"`
+
+	// IFdetails indicates the protocol and format expected by the data by the Data Scientist's application
+	// +required
+	IFdetails InterfaceDetails `json:"ifDetails"`
+
+	// Recurrence indicates how often the copy should take place or if streamed input is needed
+	// +optional
```

Comment:
You may want to add a comment about default being None.

Reply:
done